.github/workflows/ci.yml

```diff
@@ -345,6 +345,37 @@ jobs:
         run: rush build -o calling
       - name: Tests
         run: rush test -o calling
+      # Not check in the bundles stats if size doesn't change
+      # Stats size in the file is a bit random, so we can't only depend on file itself
+      - name: Check size change
+        id: changescheck
+        if: ${{ matrix.flavor == 'beta' && github.event_name == 'pull_request' }}
+        run: |
+          base_size=$(jq -r ".[0].parsedSize" samples/Calling/bundle-report/report.json)
+          current_size=$(jq -r ".[0].parsedSize" samples/Calling/dist/build/report.json)
+          change=$([ $current_size = $base_size ] && echo "true" || echo "false")
+          echo "::set-output name=hasChanged::${change}"
+      - name: Copy current bundle report json
+        if: ${{ steps.changescheck.outputs.hasChanged == 'true' }}
+        run: cp samples/Calling/dist/build/report.json samples/Calling/bundle-report/
+      # Setup bot information for pushing new changes
+      # Here we use the id from the github actions bot: https://api.github.com/users/better-informatics%5Bbot%5D
+      - name: Setup bot git information
+        if: ${{ steps.changescheck.outputs.hasChanged == 'true' }}
+        run: |
+          git config user.email "41898282+github-actions[bot]@users.noreply.github.com"
+          git config user.name "github-actions[bot]"
+      - name: Commit bundle report
+        if: ${{ steps.changescheck.outputs.hasChanged == 'true' }}
+        env:
+          # use GITHUB_TOKEN so it won't trigger recursive PR validation
+          GITHUB_TOKEN: ${{secrets.GITHUB_TOKEN}}
+        run: |
+          git checkout ${{ github.event.pull_request.head.ref }}
```

Comment:
note: is this isn't a pull request presumably this ref won't exist - e.g. Ci on main branch, or if action run directly from workflow_dispatch

Reply:
never mind the if checks for pull requets!